### ISSUE TITLE
Error when using branches with floating numeric names

### ISSUE THREAD

**Issue by fernando-pcg** [peter-evans/create-pull-request#3493]
Error when using branches with floating numeric names

In my projects, branch names of the floating type are used, for example 15.0, 16.0, but the parameters do not allow this, when using these numbers it does not consider the .0 and it is not possible to change these names, so it does not find the branch, I also tried to put these names inside quotes but it gives another error.

### Steps to reproduce
<img width="733" alt="image" src="https://github.com/user-attachments/assets/876123b4-cf70-45b8-86f6-f5d412c49d48">
Create the action with a branch with a floating name.

If this issue is describing a possible bug please provide (or link to) your GitHub Actions workflow.
<img width="960" alt="image" src="https://github.com/user-attachments/assets/256304dc-3c96-4da0-9900-e566fc748ed6">

**Comment by peter-evans** [comment]
Hi @fernando-pcg 

This might be an issue with the yaml parsing on the runner. Could you try setting the base like this and see if it fixes it:

```yml
base: "16.0"
```

**Comment by peter-evans** [comment]
> I also tried to put these names inside quotes but it gives another error

I just saw you already tried that. Could you share that error/log.

**Comment by peter-evans** [comment]
I've spotted another issue, which is probably the "another error" you mentioned. Setting the `branch` like this will not work.

```yml
branch: ${{ github.ref_name }}
```

Please see here for an explanation: https://github.com/peter-evans/create-pull-request/blob/main/docs/common-issues.md#create-using-an-existing-branch-as-the-pr-branch

**Comment by fernando-pcg** [comment]
> I've spotted another issue, which is probably the "another error" you mentioned. Setting the `branch` like this will not work.
> 
> ```yaml
> branch: ${{ github.ref_name }}
> ```
> 
> Please see here for an explanation: https://github.com/peter-evans/create-pull-request/blob/main/docs/common-issues.md#create-using-an-existing-branch-as-the-pr-branch

Yes, I realized that it didn't work, so I didn't try to use it again.

**Comment by fernando-pcg** [comment]
> Hi @fernando-pcg
> 
> This might be an issue with the yaml parsing on the runner. Could you try setting the base like this and see if it fixes it:
> 
> ```yaml
> base: "16.0"
> ```
![image](https://github.com/user-attachments/assets/95788b59-0d4c-4107-9087-ec628b3fad93)
This is the error I get when trying with '16.0', 
 I can try it with the double quotes, but I guess it would be something similar, I'll try it anyway

**Comment by peter-evans** [comment]
@fernando-pcg Using quotes is the correct way to avoid the original error you had. So please keep the quotes. 👍 

The problem now is what I mentioned in this comment: https://github.com/peter-evans/create-pull-request/issues/3493#issuecomment-2457813184

**Comment by fernando-pcg** [comment]
> @fernando-pcg Using quotes is the correct way to avoid the original error you had. So please keep the quotes. 👍
> 
> The problem now is what I mentioned in this comment: [#3493 (comment)](https://github.com/peter-evans/create-pull-request/issues/3493#issuecomment-2457813184)

Get it, im gonna look for a work around
